tmt/lint.py

```diff
@@ -0,0 +1,238 @@
+
+       if not self.path:
+           yield LinterOutcome.FAIL, 'directory path is not set'
+           return
+
+       test_path = os.path.join(self.node.root, os.path.relpath(self.path.strip(), '/'))
+
+       if not os.path.exists(test_path):
+           yield LinterOutcome.FAIL, f'test path "{test_path}" does not exist'
+           return
+
+       yield LinterOutcome.PASS, f'test path "{test_path}" does exist'
+
+   def lint_manual_valid_markdown(self) -> LinterReturn:
+       ''' T008: manual test should be valid markdown '''
+
+       # Linter should yield `SKIP` outcome when it does not apply, to announce
+       # it did inspect the object but realized it's out of scope.
+       if not self.manual:
+            yield LinterOutcome.SKIP, 'not a manual test'
+            return
+
+       manual_test = os.path.join(self.node.root, self.path.strip())
+
+       warnings = tmt.export.check_md_file_respects_spec(manual_test)
+
+       if warnings:
+           # Linter may yield as many tuples as deems necessary. This allows for
+           # linters iterating over more granular aspects of metadata, providing
+           # more specific hints.
+           for warning in warnings:
+              yield LinterOutcome.WARN, warning
+
+       ...
+"""
+
+import dataclasses
+import enum
+import textwrap
+from typing import (TYPE_CHECKING, Any, Callable, ClassVar, Generator, Generic,
+                    Iterable, List, Optional, Tuple, TypeVar)
+
+from click import style
+
+import tmt
+import tmt.utils
+
+if TYPE_CHECKING:
+    import tmt.base
+
+
+# ignore[type-arg]: bound type vars cannot be generic, and it would create a loop anyway.
+LintableT = TypeVar('LintableT', bound='Lintable')  # type: ignore[type-arg]
+
+
+class LinterOutcome(enum.Enum):
+    SKIP = 'skip'
+    PASS = 'pass'
+    FAIL = 'fail'
+    WARN = 'warn'
+    FIXED = 'fixed'
+
+
+# TODO: these would be cool, but first we should get rid of Result
+# adding color when it shouldn't have.
+#
+# _OUTCOME_TO_MARK = {
+#     LinterOutcome.SKIP: '-',
+#     LinterOutcome.PASS: '\N{heavy check mark}',
+#     LinterOutcome.FAIL: '\N{heavy multiplication x}',
+#     LinterOutcome.WARN: '\N{exclamation mark}',
+#     LinterOutcome.FIXED: '\N{heavy check mark}'
+# }
+
+_OUTCOME_TO_MARK = {
+    LinterOutcome.SKIP: 'skip',
+    LinterOutcome.PASS: 'pass',
+    LinterOutcome.FAIL: 'fail',
+    LinterOutcome.WARN: 'warn',
+    LinterOutcome.FIXED: 'fix '
+    }
+
+_OUTCOME_TO_COLOR = {
+    LinterOutcome.SKIP: 'green',
+    LinterOutcome.PASS: 'green',
+    LinterOutcome.FAIL: 'red',
+    LinterOutcome.WARN: 'yellow',
+    LinterOutcome.FIXED: 'green'
+    }
+
+
+#: Info on how a linter decided: linter itself, its outcome & the message.
+LinterRuling = Tuple['Linter', LinterOutcome, str]
+#: A return value type of a single linter.
+LinterReturn = Generator[Tuple[LinterOutcome, str], None, None]
+#: A linter itself, a callable method.
+LinterCallback = Callable[['Lintable'], LinterReturn]
```

Comment:
Variables used only for type annotations usually have `Type` suffix?
```suggestion
LinterRulingType = Tuple['Linter', LinterOutcome, str]
#: A return value type of a single linter.
LinterReturnType = Generator[Tuple[LinterOutcome, str], None, None]
#: A linter itself, a callable method.
LinterCallbackType = Callable[['Lintable'], LinterReturnType]
```

Reply:
That's a question. Recently I tend towards not using the `Type` suffix, as the "type" is sort of already implied - `-> Foo` means "the function returns a value of type `Foo`", therefore I started to consider `FooType` as superfluous.